source/ScreensaverHero.hx

```diff
@@ -0,0 +1,61 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.math.FlxPoint;
+import flixel.math.FlxRandom;
+import flixel.util.FlxColor;
+
+class ScreensaverHero extends FlxSprite
+{
+	var randomSeed:FlxRandom;
+	var xDirectionIndicator:Int;
+	var yDirectionIndicator:Int;
+
+	public function new(X:Float = 0, Y:Float = 0)
+	{
+		super(X, Y);
+		this.makeGraphic(40, 20, FlxColor.BLUE, false);
+		xDirectionIndicator = Std.int(FlxG.random.int(-1, 1));
+		yDirectionIndicator = Std.int(FlxG.random.int(-1, 1));
+
+		if (xDirectionIndicator == 0 && yDirectionIndicator == 0)
+		{
+			// now using awesome built in number generator!
+			xDirectionIndicator = FlxG.random.sign(50);
+			yDirectionIndicator = FlxG.random.sign(50);
+		}
+	}
+
+	override public function update(elapsed:Float)
+	{
+		this.setColorTransform(xDirectionIndicator, yDirectionIndicator, xDirectionIndicator, 1, Std.int(x % 255), Std.int(y % 255), Std.int(x % 255), 1);
+		this.setPosition(x + xDirectionIndicator, y + yDirectionIndicator);
```

Comment:
Oh, side thing that I forgot to point out on the first go-around: Since we're changing this object's position by a set amount every time update() is called, we're doing the old-school thing where our game logic is framerate-dependent. If someone's computer ran the game at 50 fps instead of 60, we'd things moving at 5/6 of the speed.

Using the engine's built-in velocity is the way to disconnect the physics from framerate (which you already do in the ControllableHero class). Wanted to bring it up because it was on my mind, but it's not something that you have to change for this assignment.

I'll just leave it up to you 👍

Reply:
Yeah when I found velocity and drag I was like "This is amazing!"